storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -1445,9 +1464,10 @@ bool Redo_Log_Data_Manager::stop_at(lsn_t lsn, lsn_t checkpoint_lsn) {
   archived_log_monitor.stop();
 
   /* to ensure redo logs are not disabled during the backup, reopen the log
-  files to read HEADER */
-  if (!reopen_log_files(lsn)) {
-    xb::error() << "Cannot find file with LSN " << lsn;
+  files to read HEADER. */
+  if (opt_lock_ddl == false && archived_log_state == ARCHIVED_LOG_NONE &&
+      !reopen_log_files(0)) {
+    xb::error() << "Failed to open redo log files. ";
     return (false);
   }
   log_crash_safe_validate(*log_sys);
```

Comment:
@rahulmalik87 can redo log be disabled after backup is started?

Reply:
I have tested both PS(LOCK tables for backup ) and MS (Lock instance for backup) . 
Redo log can not be disabled with LTFB or LIFB. But if lock-ddl=false they can be disabled. 

Also if log archive is set we can disable redo archive. 

so it can be disabled if not using lock-ddl and archive.

Reply:
ok, there is one case that will cause an assertion (existing issue)

1. redo log is enabled at the time of backup
2. backup is taken with --lock-ddl=OFF
3. redo is disabled
4. at the end of backup, we reach log_crash_safe_validate()

log_crash_safe_validate() has a ut_a() assertion that checks that redo should be enabled.

Reply:
At step 4, It will fail to reopen_log_files. We will not reach log_crash_safte_validate.

Reply:
ok, if we do validate check as part of reopen, all good